frontend/js/src/explore/artist-similarity/Data.tsx

```diff
@@ -0,0 +1,208 @@
+    if (artistsData.length) {
+      // Get the main artist from the first dataset
+      setMainArtist(artistsData[0].data[0]);
+      // Get the similar artists from the second dataset
+      const similarArtistsResponse = artistsData[1];
+      if (similarArtistsResponse.data.length) {
+        setCompleteSimilarArtistsList(similarArtistsResponse.data);
+      }
+      // In case no similar artists are found
+      else {
+        setCompleteSimilarArtistsList([]);
+      }
+    }
+    setColors((prevColors) => [prevColors[1], prevColors[1].tetrad()[1]]);
+  }, []);
+
+  const transformedArtists: GraphDataType = mainArtist
+    ? generateTransformedArtists(
+        mainArtist,
+        similarArtistsList,
+        colors[0],
+        colors[1],
+        similarArtistsLimit
+      )
+    : {
+        nodes: [],
+        links: [],
+      };
+
+  const fetchData = useCallback(
+    async (artist_mbid: string): Promise<void> => {
+      try {
+        const response = await fetch(BASE_URL + artist_mbid);
+        const data = await response.json();
+        processData(data);
+      } catch (error) {
+        setSimilarArtistsList([]);
+        toast.error(
+          <ToastMsg
+            title="Search Error"
+            message={typeof error === "object" ? error.message : error}
+          />,
+          { toastId: "error" }
+        );
+      }
+    },
+    [processData]
+  );
+
+  // Update the graph when either artistMBID or similarArtistsLimit changes
+  useEffect(() => {
+    fetchData(artistMBID);
+  }, [artistMBID, fetchData]);
+  // Update the graph when limit changes by only changing the data and not making a new request to server
+  useEffect(() => {
+    const newSimilarArtistsList = completeSimilarArtistsList.slice(
+      0,
+      similarArtistsLimit
+    );
+    setSimilarArtistsList(newSimilarArtistsList);
+  }, [completeSimilarArtistsList, similarArtistsLimit]);
+
+  useEffect(() => {
+    window.postMessage(
+      { brainzplayer_event: "current-listen-change", payload: currentTracks },
+      window.location.origin
+    );
+    window.postMessage(
+      { brainzplayer_event: "play-listen", payload: currentTracks },
+      window.location.origin
+    );
+  }, [currentTracks]);
+
+  const backgroundColor1 = colors[0]
+    .clone()
+    .setAlpha(BACKGROUND_ALPHA)
+    .toRgbString();
+  const backgroundColor2 = colors[1]
+    .clone()
+    .setAlpha(BACKGROUND_ALPHA)
+    .toRgbString();
+  const backgroundGradient = `linear-gradient(${
+    Math.random() * 360
+  }deg ,${backgroundColor1},${backgroundColor2})`;
+  const { APIService, currentUser } = React.useContext(GlobalAppContext);
+  return (
+    <div className="artist-similarity-main-container">
+      <SearchBox
+        onArtistChange={setArtistMBID}
+        onSimilarArtistsLimitChange={setSimilarArtistsLimit}
+        currentsimilarArtistsLimit={similarArtistsLimit}
+      />
+      <div className="artist-similarity-graph-panel-container">
```

Comment:
There should probably be a helpful message displayed on screen if the user hasn't yet selected an artist (cf. comment above about not having a default hardcoded)